atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/Schema.java

```diff
@@ -292,10 +292,10 @@ public void validate() {
                         "In index, a component derived from column must reference an existing column");
             }
 
-            if (indexMetadata.getIndexType().equals(IndexType.CELL_REFERENCING)) {
+            if (!indexMetadata.getIndexType().equals(IndexType.ADDITIVE)) {
```

Comment:
This condition is equivalent, but now it:
- Matches the framing of the error message.
- Is restrictive by default, providing additional protection if we introduce a new `IndexType` in the future.